File: Chapter 04/Java/src/main/java/com/grokkingsoftwarearchitecture/chapter04/section_4_4_anti_patterns/before_fat_controller_anemic_domain/Order.java
package com.grokkingsoftwarearchitecture.chapter04;

import java.util.List;

/**
 * THE ANEMIC DOMAIN MODEL.
 * ARCHITECTURE WARNING: This is just a data container #A.
 * It has no business logic, making it "Anemic" #B.
 */
public class Order {
    public int id;
    public double total;
    public String customerEmail;
    public List<Item> items;
}

File: Chapter 04/JS/section_4_4_anti_patterns/before_fat_controller_anemic_domain/order.js
/**
 * THE ANEMIC DOMAIN MODEL.
 * Just a property holder with no internal rules #A, #B.
 */
class Order {
    constructor() {
        this.id = null;
        this.total = 0;
        this.customerEmail = "";
        this.items = [];
    }
}

module.exports = Order;